### PR TITLE
Add Trexle gateway

### PR DESCRIPTION
**Pull request by hossamhossny**

### CODE REVIEW

**Comment by akDeveloper** [comment]
Much better now.
I see you made the PR to master branch. I prefer PR against develop branch, but if this is causing the last strange behavior, i 'll merge your changes to the develop branch.

Also i would like to see some tests for this new gateway.

File: lib/AktiveMerchant/Billing/Gateways/Trexle.php [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+ *        "verification_value" => "123"
+ *  ]); // Card data
+ *
+ *  // Complete purchase
+ *  $response = $gateway->purchase(1000, $card);
+ *
+ *  // Get unique reference from transaction for future use.
+ *  $authorization = $response->authorization();
+ *
+ * @author Andreas Kollaros <andreas@larium.net>
```
Comment (akDeveloper) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: You should change this with your name

Comment (hossamhossny) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: It's actually your own code for Pin. Trexle API is identical, so I preferred to keep your name.
But if you insist, sure! 👍

File: lib/AktiveMerchant/Billing/Gateways/Trexle.php [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * If you want to unstore a credit card you should also keep card token from
+     * response `$response->card->token`
+     * Then pass it as option in unstore method below.
+     */
+    public function store(CreditCard $creditcard, $options = array())
```
Comment (akDeveloper) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: These doc comments applies to Stripe gateway. Does this apply to Trexle gateway too?

Comment (hossamhossny) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: Yes, Trexle supports the same thing too here.

**Comment by hossamhossny** [comment]
I think the strangeness happened when I pushed into the develop branch from my develop branch. I pushed from master to master to see what happens, and it worked.

File: lib/AktiveMerchant/Billing/Gateways/Trexle.php (hunk 1 above)
Comment (akDeveloper) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: Yes, i prefer the contributor name. Thanks

File: lib/AktiveMerchant/Billing/Gateways/Trexle.php (hunk 2 above)
Comment (hossamhossny) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: Looks all good to me know. Do you want me to add a /tests/AktiveMerchant/Billing/Gateways/TrexleTest.php file?

Comment (hossamhossny) [inline lib/AktiveMerchant/Billing/Gateways/Trexle.php]: I added a test unit

File: tests/AktiveMerchant/Billing/Gateways/TrexleTest.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    {
+        $this->mock_request($this->successCreditResponse());
+        $response = $this->gateway->credit(
+            $this->amount,
+            $authorization
+        );
+
+        $this->assert_success($response);
+        $this->assertNotNull($response->authorization());
+        $this->assertEquals('Pending', $response->message());
```
Comment (akDeveloper) [inline tests/AktiveMerchant/Billing/Gateways/TrexleTest.php]: It doesn't seem to create a `Pending` message string and this test is failing.

**Comment by akDeveloper** [comment]
I created a new PR #70 and rebased your changes.
I will merge that PR.

Thanks for your contribution and good luck with your payment gateway :+1:

**Pull request closed without merging**